modules/mod_comment/mod_comment.erl

```diff
@@ -94,20 +105,36 @@ observe_search_query(_, _Context) ->
 %% @doc Move all comments from one resource to another
 observe_rsc_merge(#rsc_merge{looser_id=LooserId, winner_id=WinnerId}, Context) ->
     m_comment:merge(WinnerId, LooserId, Context).
+    
+observe_comment_insert(#comment_insert{comment_id=Id}, Context) ->
+    Comment = m_comment:get(Id, Context),
+    case proplists:get_value(rating, Comment) of
+        undefined -> ok;
+        _Rating ->
+           {rsc_id, RscId} = proplists:lookup(rsc_id, Comment),
+           update_rating_pivot(RscId, z_acl:sudo(Context))
+    end.
+    
+update_rating_pivot(RscId, Context) ->
+    Ratings = [R || R <- [proplists:get_value(rating, C) || C <- m_comment:list_rsc(RscId, Context)], R =/= undefined],
+    Average = lists:sum(Ratings) / length(Ratings),
+    {ok, _} = z_pivot_rsc:update_custom_pivot(RscId, {?MODULE, [{rating_average, Average}]}, Context),
+    ok.
+
 
 %% @doc Check the installation of the comment table. A bit more complicated because 0.1 and 0.2 had a table
```

Comment:
Line 125 is too long: %% @doc Check the installation of the comment table. A bit more complicated because 0.1 and 0.2 had a table.